Assets/Scripts/PlayerMovement/CharacterController.cs

```diff
@@ -35,47 +43,81 @@ void Update()
             CheckInteraction();
         }
 
-        // Character movement
-        movement.x = Input.GetAxisRaw("Horizontal");
-        movement.y = Input.GetAxisRaw("Vertical");
+        if (!controlMethod.usePhone)
+        {
+            leftStick.SetActive(false);
+            // Character movement
+            movement.x = Input.GetAxisRaw("Horizontal");
+            movement.y = Input.GetAxisRaw("Vertical");
 
-        animator.SetFloat("Horizontal", movement.x);
-        animator.SetFloat("Vertical", movement.y);
-        animator.SetFloat("Speed", movement.sqrMagnitude);
+            // Set idle position to last known direction
+            if (movement.x > 0 || movement.y > 0 || movement.x < 0 || movement.y < 0)
+            {
+                animator.SetFloat("LastHorizontal", Input.GetAxisRaw("Horizontal"));
+                animator.SetFloat("LastVertical", Input.GetAxisRaw("Vertical"));
 
-        if (movement.x == 0 || movement.y == 0)
-        {
-            footsteps.mute = true;
+                footsteps.mute = false;
+            }
         }
 
-        // Set idle position to last known direction
-        if (movement.x != 0 || movement.y != 0)
+        if (controlMethod.usePhone)
         {
-            animator.SetFloat("LastHorizontal", Input.GetAxisRaw("Horizontal"));
-            animator.SetFloat("LastVertical", Input.GetAxisRaw("Vertical"));
+            leftStick.SetActive(true);
+            if (joystick.Horizontal >= .2f || joystick.Horizontal <= -.2f)
+                movement.x = joystick.Horizontal;
+            else
+                movement.x = 0;
+
+            if (joystick.Vertical >= .2f || joystick.Vertical <= -.2f)
+                movement.y = joystick.Vertical;
+            else
+                movement.y = 0;
+
+            // Set idle position to last known direction
+            if (movement.x != 0 || movement.y != 0)
+            {
+                
+                animator.SetFloat("LastHorizontal", joystick.Horizontal);
+                animator.SetFloat("LastVertical", joystick.Vertical);
 
-            footsteps.mute = false;
+                footsteps.mute = false;
+            }
```

Comment:
this code appears to be mostly the same as above if you aren't  using a phone so maybe it could be refactored here to remove duplication of code. This will help with bug fixing later as you may fix one area but miss the other. Having it reduced to one code section will hopefully reduce bugs.